src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1361,29 +1362,43 @@ private void DecodeCustomAttributeParameterTypeOrThrow(ref BlobReader sigReader,
 
                 case SignatureTypeCode.TypeHandle:
                     // The type of the parameter can either be an enum type or System.Type.
-                    bool isNoPiaLocalType;
                     type = GetSymbolForTypeHandleOrThrow(sigReader.ReadTypeHandle(), out isNoPiaLocalType, allowTypeSpec: true, requireShortForm: true);
+                    break;
 
-                    var underlyingEnumType = GetEnumUnderlyingType(type);
-
-                    // Spec: If the parameter kind is an enum -- simply store the value of the enum's underlying integer type.
-                    if ((object)underlyingEnumType != null)
+                case SignatureTypeCode.GenericTypeParameter:
```

Comment:
>case SignatureTypeCode.GenericTypeParameter: [](http://example.com/codeflow?start=16&length=44)

Do we ever hit this case for a success scenario?

Reply:
Yes, for example in test `Attribute_GenericParameter_NestedClass`.